docs/docs/reference/02-yaml-syntax.md

```diff
@@ -23,59 +23,402 @@ hardware:
+component_a:
   ...
+
+# With the mapping directive, the node name becomes /my_component_namespace/my_new_component_name
+component_b:
+  mapping:
+    name: my_new_component_name
+    namespace: my_component_namespace
+```
+
+### Parameters
+
+The `parameters` field allows initial parameters values to be set using a `name: value` syntax.
+Currently, only string and double parameters are supported. These values are only applied when the component
+is loaded and are not dynamically reconfigurable.
+
+```yaml
+my_component:
+  parameters:
+    my_string_parameter: "my_string_value"
+    my_double_parameter: 2.0
 ```
 
-## On start
+#### Component rate
+
+The `rate` parameter is a special reserved parameter that defines the step rate of a component in Hertz, which is
+the inverse of the execution period.
+
+For example, if an image processing component should run some computation at 20 frames per second, then the
+rate parameter should be set to 20 Hertz.
+
+```yaml
+my_component:
+  parameters:
+    rate: 20
+```
+
+### Inputs and outputs
+
+The `inputs` and `outputs` fields are used to connect component signals together to enable communication, signal
+processing and control loops. Each signal is specified using a `name: value` syntax, where the name is the name
+of the signal according to the component description, and the value is and the name of the signal topic.
+If a component output is assigned to the same topic name as another component input, they are connected, as
+illustrated in the example below.
+
+```yaml
+my_component:
+  inputs:
+    robot_state: "/state"
+    applied_force: "/force"
+  outputs:
+    robot_command: "/command"
+
+my_other_component:
+  outputs:
+    force_torque_sensor: "/force"
+```
+
+### Events
+
+<!-- TODO: define the syntax for each event in more detail -->
+
+Events drive the emergent behaviour of an application. Define events to be triggered by a predicate by listing them
+under the predicate name, as shown below.
+
+```yaml
+my_component:
+  events:
+    is_active:
+      <triggered event_a>: ...
+      <triggered event_b>: ...
+    some_other_predicate_name:
+      <triggered event_c>: ...
+      <triggered event_bd>: ...
+```
+
+Read more about [events in the Concepts guide](../concepts/05-building-blocks/02-events.md).
+
+The following events are defined.
+
+#### Load or unload a component
+
+Components can be loaded or unloaded by component name.
+
+```yaml
+load:
+  component: <component_name>
+unload:
+  component: <component_name>
+```
+
+It is possible to load or unload multiple components simultaneously by specifying a list of components.
+
+```yaml
+load:
+  - component: component_a
+  - component: component_b
+```
+
+#### Transition from one component to another
+
+Component A can invoke a transition to component B as a shorthand for "unload component A, load component B".
+
+```yaml
+transition: <component_name>
+```
+
+#### Trigger a lifecycle transition
+
+```yaml
+lifecycle: "configure"
+```
```

Comment:
In my humble opinion, lines 222 to 224 should be removed, its more confusing than anything else no?

Reply:
Yeah true, it's a bit weird that it's up here before the text, probably a copy mistake